pkg/apis/build/v1alpha2/build_pod.go

```diff
@@ -189,24 +189,114 @@ func (b *Build) BuildPod(images BuildPodImages, buildContext BuildContext) (*cor
 		MountPath: sourceVolume.MountPath,
 		SubPath:   b.Spec.Source.SubPath, // empty string is a nop
 	}
-
+	platformAPILessThan07 := platformAPI.LessThan(semver.MustParse("0.7"))
 	var genericCacheArgs []string
+	var analyzerCacheArgs []string = nil
 	var exporterCacheArgs []string
 	var cacheVolumes []corev1.VolumeMount
 
 	if b.Spec.NeedVolumeCache() && buildContext.os() != "windows" {
 		genericCacheArgs = []string{"-cache-dir=/cache"}
 		cacheVolumes = []corev1.VolumeMount{cacheVolume}
+		if platformAPILessThan07 {
+			analyzerCacheArgs = genericCacheArgs
+		}
 		exporterCacheArgs = genericCacheArgs
 	} else if b.Spec.NeedRegistryCache() {
 		useCacheFromLastBuild := (b.Spec.LastBuild != nil && b.Spec.LastBuild.Cache.Image != "")
 		if useCacheFromLastBuild {
 			genericCacheArgs = []string{fmt.Sprintf("-cache-image=%s", b.Spec.LastBuild.Cache.Image)}
 		}
+		analyzerCacheArgs = genericCacheArgs
 		exporterCacheArgs = []string{fmt.Sprintf("-cache-image=%s", b.Spec.Cache.Registry.Tag)}
 	} else {
 		genericCacheArgs = nil
 	}
+
+	analyzeContainer := corev1.Container{
+		Name:      "analyze",
+		Image:     b.Spec.Builder.Image,
+		Command:   []string{"/cnb/lifecycle/analyzer"},
+		Resources: b.Spec.Resources,
+		Args: args([]string{
+			"-layers=/layers",
+			"-analyzed=/layers/analyzed.toml"},
+			analyzerCacheArgs,
+			func() []string {
+				if platformAPILessThan07 {
+					return []string{
+						"-group=/layers/group.toml",
+					}
+				}
+				return []string{}
+			}(),
+			func() []string {
+				if platformAPILessThan07 {
+					return []string{}
+				}
+				tags := []string{}
+				for _, tag := range b.Spec.Tags {
+					tags = append(tags, "-tag="+tag)
+				}
+				return tags
+			}(),
+			func() []string {
+				if b.Spec.LastBuild != nil && b.Spec.LastBuild.Image != "" {
```

Comment:
This is providing the "previous image" and falling back to the destination tag if the previous image is not available. With platform api 0.7+ and the addition of the previous-image flag we should provide the previous image explicitly.